### ISSUE TITLE
Keras slicing results in shapes must be equal error in nncase

### ISSUE THREAD

**Issue by krishnak** [kendryte/nncase#91]
I am trying to create a new model in Keras, based on an existing caffe model. The caffe model's last layer output is [N,50,50,2] which has softmax activation. After this it performs a slice to provide the top  [Nx50x50x1] as inference

My query is

1. The last layer is performing a 1x1 convolution with 2 filters to get that output, instead of that if it can be changed to perform a 1x1 convolution, it will result in an output [N,50,50,1]. Why is this not done, what are the drawbacks of doing 1x1 convolution with just 1 filter.

2. If I implement a slice operation using Keras, the output becomes Nx50x50 and the code compiles. It gets converted by tensorflowlite. However nncase is throwing an error saying "shapes must be same"

Can you please explain if slicing should be avoided or should I implement it in a better way?

**Comment by sunnycase** [comment]
Slice is supported. This should be a bug, can you attach the tflite?

**Comment by krishnak** [comment]
```
Model: "model_1"
__________________________________________________________________________________________________
Layer (type)                    Output Shape         Param #     Connected to                     
==================================================================================================
input_1 (InputLayer)            (None, 240, 320, 3)  0                                            
__________________________________________________________________________________________________
zero_padding2d_1 (ZeroPadding2D (None, 242, 322, 3)  0           input_1[0][0]                    
__________________________________________________________________________________________________
conv2d_1 (Conv2D)               (None, 120, 160, 64) 1792        zero_padding2d_1[0][0]           
__________________________________________________________________________________________________
zero_padding2d_2 (ZeroPadding2D (None, 122, 162, 64) 0           conv2d_1[0][0]                   
__________________________________________________________________________________________________
conv2d_2 (Conv2D)               (None, 60, 80, 64)   36928       zero_padding2d_2[0][0]           
__________________________________________________________________________________________________
conv2d_3 (Conv2D)               (None, 60, 80, 64)   36928       conv2d_2[0][0]                   
__________________________________________________________________________________________________
conv2d_4 (Conv2D)               (None, 60, 80, 64)   36928       conv2d_3[0][0]                   
__________________________________________________________________________________________________
add_1 (Add)                     (None, 60, 80, 64)   0           conv2d_2[0][0]                   
                                                                 conv2d_4[0][0]                   
__________________________________________________________________________________________________
conv2d_5 (Conv2D)               (None, 60, 80, 64)   36928       add_1[0][0]                      
__________________________________________________________________________________________________
conv2d_6 (Conv2D)               (None, 60, 80, 64)   36928       conv2d_5[0][0]                   
__________________________________________________________________________________________________
add_2 (Add)                     (None, 60, 80, 64)   0           add_1[0][0]                      
                                                                 conv2d_6[0][0]                   
__________________________________________________________________________________________________
conv2d_7 (Conv2D)               (None, 60, 80, 64)   36928       add_2[0][0]                      
__________________________________________________________________________________________________
conv2d_8 (Conv2D)               (None, 60, 80, 64)   36928       conv2d_7[0][0]                   
__________________________________________________________________________________________________
add_3 (Add)                     (None, 60, 80, 64)   0           add_2[0][0]                      
                                                                 conv2d_8[0][0]                   
__________________________________________________________________________________________________
conv2d_9 (Conv2D)               (None, 60, 80, 128)  8320        add_3[0][0]                      
__________________________________________________________________________________________________
conv2d_10 (Conv2D)              (None, 60, 80, 128)  16512       conv2d_9[0][0]                   
__________________________________________________________________________________________________
conv2d_11 (Conv2D)              (None, 60, 80, 2)    258         conv2d_10[0][0]                  
__________________________________________________________________________________________________
lambda_1 (Lambda)               (None, 60, 80)       0           conv2d_11[0][0]                  
__________________________________________________________________________________________________
conv2d_14 (Conv2D)              (None, 60, 80, 128)  16512       conv2d_9[0][0]                   
==================================================================================================
Total params: 301,890
Trainable params: 301,890
Non-trainable params: 0
__________________________________________________________________________________________________
None
```

The above is Keras output for the model. Attached is the tflite

I get the following error message when trying to compile

```
ncc compile amodel.tflite amodel.kmodel -i tflite --dataset ~/nncase/images/1.jpg --dump-ir
1. Import graph...
2. Optimize Pass 1...
Fatal: Shapes must be same, but got [1x60x80] and [1x60x80x1]
```
[amodel.zip](https://github.com/kendryte/nncase/files/4184255/amodel.zip)

**Comment by krishnak** [comment]
My lambda for slicing is as follows,

`slicedoutputfaces = Lambda(lambda x: x[:,:,:,0])(outputfaces)`

**Comment by sunnycase** [comment]
Fixed in a17b3f5

**Comment by krishnak** [comment]
Fix Works, Thanks.
